### PR TITLE
Add EarlyStopping integration to TensorFlow.Keras autologging

### PR DESCRIPTION
**Pull request by juntai-zheng**
## What changes are proposed in this pull request?

Mirrors work on #2219 by adding EarlyStopping callback integration to TensorFlow.Keras (both 1.X and 2.X). Does not include `fit_generator()` support.

## How is this patch tested?

Unit tests written in `tests/tensorflow_autolog/test_tensorflow_autolog.py` and `tests/tensorflow_autolog/test_tensorflow2_autolog.py`

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Same as #2219 , but this is for TensorFlow.Keras.

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [X] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by smurching — COMMENTED** [review]
Mostly looks good. We should decide what to do regarding the tradeoff of:
1. Log both loss & epoch_loss, accuracy & epoch_acc metrics, including in the non-early stopping case (the current behavior - IMO this might confuse users). We could make this better by only logging the extra `loss` & `acc` metrics in the early-stopping case
2. Remove Tensorboard callback from Keras model to avoid generating `epoch_loss`, `epoch_acc` metrics. This has the downside of us no longer generating Tensorboard logs as an artifact, & if the user does include a TensorBoard callback we'll still log `epoch_loss` etc.
3. Provide special handling when logging metrics of the restored model after early stopping - log loss & accuracy with the `epoch_` prefix, & don't log an extra step of other metrics

Out of these options I think I like 3 the most, if it's not too complicated - let me know what you think.

**Review by smurching — APPROVED** [review]
LGTM

**Comment by juntai-zheng** [comment]
We decided it was best to stick with option number 3 for simplicity. Note that the callback logs `epoch_loss` and `epoch_acc` for TF 1.X, but `loss` and `acc` for TF 2.X.
